jujugui/static/gui/src/app/models/endpoints.js

```diff
@@ -53,127 +53,83 @@ YUI.add('juju-endpoints', function(Y) {
   },
 
   /**
-   * Find available relation targets for a service.
-   *
-   * @method getEndpoints
-   * @param {Object} application A service object.
-   * @param {Object} controller The endpoints controller.
-   *
-   * @return {Object} A mapping with keys of valid relation service targets
-   *   and values consisting of a list of valid endpoints for each.
-   */
+    Find available relation targets for an application.
+    @param {Object} application An application object.
+    @param {Object} controller The endpoints controller.
+    @return {Object} A mapping with keys of valid relation application targets
+      and values consisting of a list of valid endpoints for each.
+  */
   models.getEndpoints = function(application, controller) {
-    const targets = {},
-        requires = [],
-        provides = [],
-        appId = application.get('id'),
-        db = controller.get('db'),
-        endpointsMap = controller.endpointsMap;
+    const targets = {};
+    const appId = application.get('id');
+    const db = controller.get('db');
+    const endpointsMap = controller.endpointsMap;
+    const appEndpoints = endpointsMap[appId];
     const appIsSubordinate = application.get('subordinate');
     const appSeries = models._getSeries(db, application);
-
-    // Bail out if the map doesn't yet exist for this service.  The charm may
+    // Bail out if the map doesn't yet exist for this application. The charm may
     // not be loaded yet.
-    if (!endpointsMap[appId]) {
-      return targets;
+    if (!appEndpoints) {
+      return {};
     }
     /**
-     * Convert a service name and its relation endpoint info into a
-     * valid relation target endpoint, ie. including service name.
-     *
-     * @method getEndpoints.convert
-     */
+      Convert an application name and its relation endpoint info into a
+      valid relation target endpoint, ie. including application name.
+    */
     function convert(applicationName, relInfo) {
       return {
         service: applicationName,
         name: relInfo.name,
         type: relInfo['interface']
       };
     }
-
     /**
-     * Store endpoints for a relation to target the given service.
-     *
-     * @method getEndpoints.add
-     * @param {Object} targetEndpoint Target endpoint.
-     * @param {Object} originEndpoint Origin endpoint.
-     */
+      Store endpoints for a relation to target the given application.
+      @param {Object} targetEndpoint Target endpoint.
+      @param {Object} originEndpoint Origin endpoint.
+    */
```

Comment:
Missing `@applicationName` (I know this was an existing issue, but since you're here...)